File: 김수현/greedy/Silver/2839. 설탕 배달/설탕 배달.py
n = int(input())

result = 0

while n > 0:
    if (n % 5 == 0):
        n -= 5
        result += 1
    elif (n % 3 == 0):
        n -= 3
        result += 1
    elif (n > 5):
        n -= 5
        result += 1
    else:
        result = -1
        break

print(result)

File: 김수현/greedy/Silver/2839. 설탕 배달/설탕 배달.java
// 설탕 배달

import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int N = in.nextInt();
        int result = -1;

        in.close();

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                if (3 * i + 5 * j == N) {
                    result = i + j;
                    System.out.println(result);
                    return ;
                }
            }
        }
        System.out.println(result);
    } 
}